frontend/projects/setup-wizard/src/app/pages/embassy/embassy.page.ts

```diff
@@ -31,10 +33,14 @@ export class EmbassyPage {
     private readonly stateService: StateService,
     private readonly loadingCtrl: LoadingController,
     private readonly errorToastService: ErrorToastService,
+    private route: ActivatedRoute,
   ) {}
 
   async ngOnInit() {
     await this.getDrives()
+    this.route.queryParams.subscribe(params => {
```

Comment:
This creates a subscription that never gets unsubscribed. Instead do

`this.incomingAction = this.route.snapshot.paramMap.get('action')`

Or better yet, just do the snapshot when you need it below. ngOnInit unnecessary.